services/ScheduleService.ts

```diff
@@ -120,4 +120,122 @@ export class ScheduleService {
   formatTime(timeStr: string): string {
     return timeStr;
   }
+
+  // Helper method to determine if alternating show is active this week
+  private isAlternatingShowActive(show: ScheduleShow, targetDate: Date): boolean {
+    if (show.alternates === 0) {
+      return true; // Non-alternating show is always active
+    }
+    
+    // For alternating shows, we need a reference date to calculate weeks
+    // Using a fixed reference date of September 1, 2024 (start of fall semester)
+    const referenceDate = new Date('2024-09-01T00:00:00-04:00'); // Eastern Time
+    const targetDateEastern = new Date(targetDate.toLocaleString("en-US", {timeZone: "America/New_York"}));
+    
+    // Calculate weeks since reference date
+    const daysDiff = Math.floor((targetDateEastern.getTime() - referenceDate.getTime()) / (1000 * 60 * 60 * 24));
+    const weeksSince = Math.floor(daysDiff / 7);
+    
+    debugLog(`Show "${show.name}" alternates: ${show.alternates}, weeks since ref: ${weeksSince}, active: ${weeksSince % 2 === 0}`);
+    
+    // Even weeks = first show in alternating pair, odd weeks = second show
```

Comment:
(future) thoughts on the funky 4-show slots? not sure if we have any this season